astropy/io/ascii/core.py

```diff
@@ -665,10 +665,8 @@ def write_comments(self, lines, meta):
 
     def write(self, lines: list[str]) -> None:
         if self.start_line is not None:
-            for i, spacer_line in zip(
-                range(self.start_line), itertools.cycle(self.write_spacer_lines)
-            ):
-                lines.append(spacer_line)
+            for spacer_line in itertools.islice(itertools.cycle(self.write_spacer_lines), self.start_line):
+                lines.append(spacer_line.copy())
```

Comment:
I think this can be simplified further to an `extend()`

```suggestion
            lines.extend([line.copy() for line in itertools.islice(itertools.cycle(self.write_spacer_lines), self.start_line)])
```